R/estimate_RIT.R

```diff
@@ -0,0 +1,103 @@
+#' @title estimate RIT score for a student using regression or interpolation
+#' 
+#' @description Given studentid, measurementscale, and a target_date, the 
+#' function will return an estimated score based on selected method 
+#' 
+#' @param mapvizieR_obj mapvizieR object
+#' @param sid target studentid
+#' @param measurescale target subject
+#' @param target_date date of interest, %Y-%m-%d format
+#' @param method which method to use to estimate RIT score
+#' @param foward default is TRUE, set to FALSE if only scores before target_date should be chosen for 'closest' method
+#' 
+#' @export
+
+
+estimate_rit <- function(
+  mapvizieR_obj,
+  sid,
+  measurescale,
+  target_date,
+  method = c('closest','lm','interpolate'),
+  forward = TRUE
+) {
+  
+  # throw an error if method is not given
+  if (missing(method)) {
+    stop('method not given')
+  }
+  
+  # pull out the cdf
+  cdf <- mapvizieR_obj[['cdf']]
+  
+  target_date <- as.Date(target_date)
+  
+  # find the matchings rows
+  if (require(dplyr)) {
+    student <- cdf %>% dplyr::filter(studentid == sid,
+                                     measurementscale == measurescale)
+  } else {
+    student <- subset(cdf, studentid == sid & measurementscale == measurescale)
+  }
+  
+  # return error if student does not have take a test for given measurementscale
+  if (dim(student)[1] == 0) {
+    
+    warning('student does not have a test for given measurementscale')
+    return(NA)
+    
+  } else if (dim(student)[1] == 1) {
+    
+    warning('student only has one test event for given measurementscale')
```

Comment:
this is interesting - you're right here that you want to inform the user if they ask for non-sensical data.  I think it might make sense to put a `num_days` parameter in the function call with a default value (let's say 180?) that won't return a ritscore if it is more than `num_days` away from the requested date?  we really wouldn't want to return a ritscore that is, like, 3 years away from the requested date (_unless_ the user knows what they are doing and has set the `num_days` parameter differently)